crates/autopilot/src/solvable_orders.rs

```diff
@@ -226,6 +238,63 @@ impl SolvableOrdersCache {
     pub fn last_update_time(&self) -> Instant {
         self.cache.lock().unwrap().orders.update_time
     }
+
+    /// Quotes all limit orders and sets the fee_amount for each one to the fee returned by the
+    /// quoting process. If quoting fails, the corresponding order is filtered out.
+    async fn set_limit_order_fees(&self, orders: Vec<Order>) -> Vec<Order> {
+        stream::iter(orders.into_iter())
+            .filter_map(|mut order| async {
+                if order.metadata.class != OrderClass::Limit {
+                    return Some(order);
+                }
+
+                match self
+                    .quoter
+                    .calculate_quote(QuoteParameters {
+                        sell_token: order.data.sell_token,
+                        buy_token: order.data.buy_token,
+                        side: match order.data.kind {
+                            OrderKind::Buy => OrderQuoteSide::Buy {
+                                buy_amount_after_fee: order.data.buy_amount,
+                            },
+                            OrderKind::Sell => OrderQuoteSide::Sell {
+                                sell_amount: SellAmount::AfterFee {
+                                    value: order.data.sell_amount,
+                                },
+                            },
+                        },
+                        from: order.metadata.owner,
+                        app_data: order.data.app_data,
+                        signing_scheme: match order.signature {
+                            Signature::Eip712(_) => QuoteSigningScheme::Eip712,
+                            Signature::EthSign(_) => QuoteSigningScheme::EthSign,
+                            Signature::Eip1271(_) => QuoteSigningScheme::Eip1271 {
+                                onchain_order: false,
+                                verification_gas_limit: default_verification_gas_limit(),
+                            },
+                            Signature::PreSign => QuoteSigningScheme::PreSign {
+                                onchain_order: false,
+                            },
+                        },
+                    })
+                    .await
+                {
+                    Ok(quote) => {
+                        order.data.fee_amount = quote.fee_amount;
+                        Some(order)
+                    }
+                    Err(err) => {
+                        tracing::error!(
+                            order_uid =% order.metadata.uid, ?err,
+                            "filtered limit order due to quoting error"
+                        );
```

Comment:
Have to be careful about a catch all error log here. Error logs are serious and go to our alerts channel so if some of the errors are benign or triggerable by users doing funny things we might not want to error log them. I'm not sure if that's the case for these errors, just noting it.

Reply:
Hmm, I don't think this is something worth alerting about.

Reply:
https://github.com/cowprotocol/services/pull/690/commits/888f0270ee58f72a66eb28bf90640b0699a63e68